src/structure/spatial.jl

```diff
@@ -222,6 +253,103 @@ function psiangle(res::AbstractResidue, res_next::AbstractResidue)
     return dihedralangle(res["N"], res["CA"], res["C"], res_next["N"])
 end
 
+function psiangle(chain::Chain, res_id::Union{Integer, String})
+    inds = find(r -> resid(r)==string(res_id), chain)
+    if length(inds) != 1
+        throw(ArgumentError("$res_id is an invalid residue ID"))
+    end
+    res_list = collect(chain)
+    i = inds[1]
+    if i == length(res_list) || !sequentialresidues(res_list[i], res_list[i + 1])
+        throw(ArgumentError("$res_id is an invalid residue ID"))
+    end
+    return psiangle(res_list[i], res_list[i + 1])
+end
+
+"""
+Calculate vector of omega, psi or phi angles for a `StructuralElementOrList`.
+"""
+function omegaangles(el::StructuralElementOrList)
+    res_list = collectresidues(el, standardselector)
+    if length(res_list) < 2
+        throw(ArgumentError("At least 2 residues required to calculate dihedral angles"))
+    end
+
+    omega_angles = Float64[]
```

Comment:
This should be initiated as `Float64[NaN]` or `omegaangles[1]` will refer to the second residue in the list.